### PR TITLE
Handle selection

### LINKED ISSUE #16: Claim handle: check handle availability 2 of 2
**Description**
As a user claiming a handle, I want to know if the handle chosen is available.

**Acceptance Criteria**
1. After minimum length of the handle is met and the user stops typing for n-milliseconds
- un-grey continue button
- an icon next to the handle is displayed to represent the handle is available.
2. When the handle is not available "Continue" button is greyed.

### LINKED ISSUE #33: Don't need 'Claim this handle' button
When navigating back to the "Choose Handle" page, I should be able to see that the handle is valid and not have the next button grayed.

Screenshot 2023-08-31 at 11 31 33 AM

### PR DESCRIPTION
**Pull request by shannonwells**
# Select account and choose a handle, part 2/2
* puts wallet info into a Svelte store
* puts the handle choice into a Svelte store
* Fixes some small issues with setting state between components
* Removes the 'claim handle' button since we're saving all the signing for last

https://github.com/AmplicaLabs/wallet-proxy/assets/502640/844d5afb-f6ce-4780-9757-d4947711f7df

Closes #16
Closes #33 

To Verify:
* `npm run start`
* Go to http://localhost:5173 and select the endpoint + type that you want to test.
* Verify that you can complete all steps through Register, to get signatures for creating an MSA + delegation, and a Handle.

### CODE REVIEW

File: integration/test.ts
Comment (shannonwells) [inline integration/test.ts:1]: it was moved to e2e

File: src/lib/chain/extrinsicHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/* eslint-disable no-use-before-define */
+/* eslint-disable max-classes-per-file */
+import type { ApiPromise, ApiRx } from '@polkadot/api';
+import { firstValueFrom } from 'rxjs';
+import type { SignedBlock } from '@polkadot/types/interfaces';
+import { connect, connectPromise } from './apiConnection';
+import {isWebSocket} from "$lib/util";
+
+export class ExtrinsicHelper {
```
Comment (enddynayn) [inline src/lib/chain/extrinsicHelpers.ts:9]: Not a blocker. But how about`ChainHelper`?

**Review by enddynayn — APPROVED** [review]
👍 fantastic!

File: src/components/SelectAddress.svelte [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  import {getAccounts, onReady} from "$lib/wallet";
+  import { page } from "$app/stores";
+
+  let validAccountsArray: Array<InjectedAccount> = [];
+  let errorMessage = '';
+  export let formFinished = false;
+
+  SelectedSigningKey.subscribe((key) => {
+    if (key !== '' ) { formFinished = true;}
+  });
```
Comment (enddynayn) [inline src/components/SelectAddress.svelte:14]: I think we can use svelte derived by doing something like: 
```
$: if ($SelectedSigningKey !== '') {  formFinished = true; }
```

File: src/components/SelectAddress.svelte [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      console.error('Error: ', e.message);
+    }
+  });
+</script>
+
+{#if errorMessage !== ''}
+  <div id="error" class="text-red-600 font-xl">{errorMessage}</div>
+{:else}
+  <p class="text-2xl"><label for="signing-address">Choose an account to use</label></p>
+  <p>SigningKey: {$SelectedSigningKey}</p>
```
Comment (enddynayn) [inline src/components/SelectAddress.svelte:36]: we should also display the MSA ID associated with these addresses. But maybe out of scope.

**Comment by enddynayn** [comment]
Nice!

**Review by enddynayn — APPROVED** [review]
👍 woot woot!

**Pull request merged**